### PR TITLE
Add DDP training

### PR DESCRIPTION
**Pull request by NeelayS**

### CODE REVIEW

File: ezflow/engine/trainer.py [hunk 1]
```diff
@@ -2,16 +2,27 @@
...
 from .registry import loss_functions, optimizers, schedulers
 
 
+def seed(seed):
+
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    cudnn.benchmark = False
+    cudnn.deterministic = True
```
Comment (prajnan93) [inline ezflow/engine/trainer.py:23]: I think it would be better to set the seed for NumPy as well. 
`np.random.seed(seed)`

File: ezflow/engine/trainer.py [hunk 2]
```diff
@@ -80,19 +92,32 @@ def _setup_model(self, model):
                 device_ids = device.split(",")
                 device_ids = [int(id) for id in device_ids]
                 cuda_str = "cuda:" + device
```
Comment (prajnan93) [inline ezflow/engine/trainer.py:94]: unused variable `cuda_str`

Comment (prajnan93) [inline ezflow/engine/trainer.py:94]: `cuda_str` unused variable

File: configs/trainers/base.yaml [hunk 3]
```diff
@@ -25,8 +25,13 @@ GRAD_CLIP:
   USE: True
   VALUE: 1.0
 DEVICE: "0"
-DISTRIBUTED: False
-DISTRIBUTED_BACKEND: "nccl"
+DISTRIBUTED:
```
Comment (prajnan93) [inline configs/trainers/base.yaml:28]: let's add these config options for command line arguments in `ezflow/tools/train`

File: ezflow/data/dataloader/dataloader_creator.py [hunk 4]
```diff
@@ -361,14 +374,33 @@ def get_dataloader(self):
...
+            sampler = DistributedSampler(
+                dataset,
+                num_replicas=self.world_size,
+                rank=self.rank,
+                shuffle=self.shuffle,
+                drop_last=self.drop_last,
+            )
+            data_loader = DataLoader(
+                dataset,
+                batch_size=self.batch_size,
```
Comment (prajnan93) [inline ezflow/data/dataloader/dataloader_creator.py:388]: I think the batch size should be set based on the number of GPU devices. For example: if `batch_size` is 10 and total GPU devices are 4 then the batch_size for distributed training is 10*4=40.

File: configs/trainers/base.yaml (hunk 3 above)
Comment (NeelayS) [reply configs/trainers/base.yaml:28]: So there are such 6 such config parameters for distributed training. Do you think we should have all of them as command line arguments? Or we can just have a single flag for enabling distributed training and point the user to the config file for the other parameters?

File: ezflow/data/dataloader/dataloader_creator.py (hunk 4 above)
Comment (prajnan93) [reply ezflow/data/dataloader/dataloader_creator.py:388]: I am not entirely sure about this either. 

This is Professor Huaizu's implementation: [https://github.com/playerkk/deeplab_flow/blob/30a7a758ea75ef4164918b892ee288b71ee44820/tools/train_optical_flow.py#L204](https://github.com/playerkk/deeplab_flow/blob/30a7a758ea75ef4164918b892ee288b71ee44820/tools/train_optical_flow.py#L204)

**Review by prajnan93 — COMMENTED** [review]
Added comments for DDP.
